### PR TITLE
Type checking for `FederationHandler`

### PR DESCRIPTION
**Pull request by richvdh**
fix a few things to make this pass mypy.

### CODE REVIEW

File: synapse/handlers/federation.py [hunk 1]
```diff
@@ -2828,6 +2835,12 @@ async def _check_signature(self, event, context):
                         return
             except Exception as e:
                 last_exception = e
+
+        if last_exception is None:
+            # we can only get here if get_public_keys() returned an empty list
+            # TODO: make this better
+            raise RuntimeError("no public key in invite event")
```
Comment (clokep) [inline synapse/handlers/federation.py:2842]: Seems fine, but curious if we were actually running into it?

Comment (richvdh) [reply synapse/handlers/federation.py:2842]: I don't *think* so, but I had to do something here to keep mypy happy.

Comment (richvdh) [reply synapse/handlers/federation.py:2842]: in theory I suspect we could run into it if the invite was malformed.

Comment (clokep) [reply synapse/handlers/federation.py:2842]: Makes sense! The result is similar either way, wasn't sure if you came across it in Sentry or something. 👍

**Pull request merged**